### PR TITLE
Add dependencies from NioImapClient to dependencyManagement

### PR DESCRIPTION
**Pull request by stevegutz**
This will allow us to move `NioImapClient` to the newest version of basepom.

@jhaber 
/cc @zklapow

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -69,9 +69,13 @@
     <dep.zookeeper.version>3.4.6</dep.zookeeper.version>
 
     <dep.assertj.version>3.4.1</dep.assertj.version>
+    <dep.google.clients.version>1.20.0</dep.google.clients.version>
```
Comment (stevegutz) [inline pom.xml]: Is this safe?

File: pom.xml [hunk 2]
```diff
@@ -937,6 +1013,37 @@
...
+        <groupId>com.spotify</groupId>
+        <artifactId>completable-futures</artifactId>
+        <version>0.3.0</version>
+      </dependency>
+
+      <dependency>
+        <groupId>org.immutables</groupId>
+        <artifactId>value</artifactId>
+        <version>2.1.15</version>
+        <scope>provided</scope>
```
Comment (jhaber) [inline pom.xml]: We almost always want this at scope provided, but I'd still avoid adding a scope here

File: pom.xml (hunk 1 above)
Comment (jhaber) [inline pom.xml]: Seems ok

**Comment by jhaber** [comment]
lgtm

File: pom.xml [hunk 3]
```diff
@@ -370,7 +374,7 @@
       <dependency>
         <groupId>com.sun.mail</groupId>
         <artifactId>javax.mail</artifactId>
-        <version>1.5.2</version>
+        <version>1.5.6</version>
```
Comment (stevegutz) [inline pom.xml]: Bumped this as well

**Pull request merged**